src/status_im2/common/floating_button_page/view.cljs

```diff
@@ -0,0 +1,113 @@
+        (< (- @scroll-view-height button-container-height) @content-container-height)
+
+        platform/ios?
+        (< (- @scroll-view-height keyboard-view-height) (- @content-container-height content-scroll-y))
+
+        :else
+        false))))
+
+(defn button-container
+  [{:keys [show-keyboard? keyboard-shown show-background? keyboard-height]} children]
+  (let [insets (safe-area/get-insets)
+        height (reagent/atom 0)]
+    (reset! height (if show-keyboard? (if keyboard-shown keyboard-height 0) 0))
+    [rn/view {:style {:margin-top :auto}}
+     (cond
+       (and (> @height 0) show-background?)
+       [blur/view
+        (when keyboard-shown
+          {:blur-amount      34
+           :blur-type        :transparent
+           :overlay-color    :transparent
+           :background-color (if platform/android? colors/neutral-100 colors/neutral-80-opa-1-blur)
+           :style            (style/blur-button-container insets)})
+        children]
+
+       (and (> @height 0) (not show-background?))
+       [rn/view {:style (style/view-button-container true insets)}
+        children]
+
+       (not show-keyboard?)
+       [rn/view {:style (style/view-button-container false insets)}
+        children])]))
+
+(defn f-view
+  [{:keys [button-props button-label]} header children]
+  (reagent/with-let [show-keyboard?   (reagent/atom false)
+                     content-scroll-y (reagent/atom 0)
+                     show-listener    (oops/ocall rn/keyboard
+                                                  "addListener"
+                                                  (if platform/android?
+                                                    "keyboardDidShow"
+                                                    "keyboardWillShow")
+                                                  #(reset! show-keyboard? true))
+                     hide-listener    (oops/ocall rn/keyboard
+                                                  "addListener"
+                                                  (if platform/android?
+                                                    "keyboardDidHide"
+                                                    "keyboardWillHide")
+                                                  #(reset! show-keyboard? false))]
+
+    (let [{:keys [keyboard-shown
+                  keyboard-height]} (hooks/use-keyboard)
+          show-background?          (show-button-background keyboard-height
+                                                            keyboard-shown
+                                                            @content-scroll-y)]
+
+      [rn/view {:style style/page-container}
+       header
+       [rn/scroll-view
+        {:on-layout               (fn [event]
+                                    (let [height (oops/oget event "nativeEvent.layout.height")]
+                                      (reset! scroll-view-height height)
+                                      (reset! content-scroll-y 0)))
```

Comment:
Can you please explain why are these values needed?

Reply:
sure, I will also make these not global refs on cleanup.
The idea here is that the background of the button container only shows when it is over the content of the page. These are used to figure out that line so the scroll height knows when the button is over the content or not